community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/VersionAwareLogEntryReader.java

```diff
@@ -61,41 +65,70 @@ public LogEntry readLogEntry( SOURCE channel ) throws IOException
         try
         {
             LogPositionMarker positionMarker = new LogPositionMarker();
-            channel.getCurrentPosition( positionMarker );
+            long skipped = 0;
             while ( true )
             {
-                LogEntryVersion version = null;
-                LogEntryParser<LogEntry> entryReader;
+                channel.getCurrentPosition( positionMarker );
+
+                /*
+                 * if the read type is negative than it is actually the log entry version
+                 * so we need to read an extra byte which will contain the type
+                 */
```

Comment:
This comment would be better closer to the logic that does this checking, rather than here where the variables are declared. i.e. move it back down to where it was, or even to the if-statement.